reproducible/faiss/Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM docker.io/library/debian:11
```

Comment:
please update to debian:12. We should be using `bookworm` where its possible to do so. I understand this introduces some dependency pain. That is why we are doing this work.:)

Reply:
Done!